python/ray/train/lightning/_lightning_utils.py

```diff
@@ -226,23 +226,36 @@ class RayTrainReportCallback(pl.callbacks.Callback):
     This callback is a subclass of `lightning.pytorch.callbacks.Callback
     <https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.Callback.html#lightning.pytorch.callbacks.Callback>`_.
 
-    It fetches the latest `trainer.callback_metrics` and reports together with
-    the checkpoint on each training epoch end.
+    At the end of each training epoch, this callback dumps a checkpoint with
+    `lightning.pytorch.Trainer.save_checkpoint`, and fetches the latest metrics
+    with `trainer.callback_metrics`. It then reports the metrics and checkpoint
+    via `ray.train.report`.
+
+    This callback supports distributed checkpointing. It efficiently uploads
+    checkpoint shards from individual workers in parallel, such as those from
+    Deepspeed ZeRO and FSDP.
 
     Checkpoints will be saved in the following structure::
 
         checkpoint_00000*/      Ray Train Checkpoint
         └─ checkpoint.ckpt      PyTorch Lightning Checkpoint
 
     For customized reporting and checkpointing logic, implement your own
-    `lightning.pytorch.callbacks.Callback` following this user
+    lightning callback following this user
     guide: :ref:`Saving and Loading Checkpoints <train-dl-saving-checkpoints>`.
+
+    Args:
+        rank_zero_only: Whether to upload checkpoints from world rank
+        zero worker only. Default is False.
```

Comment:
Would we want to support local rank 0 as another option?

Reply:
That's a good idea, but for now I don't want to make it over complicated. The Beginners/DDP users might get confused about it.

From my understanding, if a users knows the behavior and benefit of `local_rank_zero_only`, he must be an expert of deepspeed/fsdp and Ray Train. I'd recommend to let these advanced users to implement their own report callback. What do you think?